### PR TITLE
test PR

### PR DESCRIPTION
**Pull request by WindzCUHK**
# Description

test PR

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

## Related issue/PR

**Delete this section if there are no issues or pull requests that relate to this pull request.**
- Fixes #_issue_
- Closes #_PR_

---

## Checklist

- [ ] Followed the guidelines in the CONTRIBUTING document
- [ ] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [ ] Tested and linted the code
- [ ] Commented the code
- [ ] Made corresponding changes to the documentation
- [ ] Passed all pipeline checking

## Checklist for maintainer
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge

### CODE REVIEW

File: .github/workflows/docker-build-publish.yaml [hunk 1]
```diff
@@ -24,6 +24,9 @@ on:
       DOCKER_REGISTRY_REF:
         description: "docker registry reference of the container image"
         value: ${{ jobs.docker-env.outputs.DOCKER_REGISTRY_REF }}
+      DOCKER_REGISTRY_PUSH:
```
Comment (ctyano) [inline .github/workflows/docker-build-publish.yaml:27]: @WindzCUHK I cannot find any necessity of adding a new environment variable just for the boolean flag.
`${{ github.event_name != 'pull_request' }}` looks more straightforward for the pipeline and adding a new variable `DOCKER_REGISTRY_PUSH` looks redundant.

Comment (mlajkim) [reply .github/workflows/docker-build-publish.yaml:27]: 👀

Comment (WindzCUHK) [reply .github/workflows/docker-build-publish.yaml:27]: @ctyano
I want to push the container image as `pr-4` in the PR during the development phrase.
As there are multiple steps using the `${{ github.event_name != 'pull_request' }}` condition, it is easier to just change the variable's value. (example commit: https://github.com/WindzCUHK/k8s-athenz-sia/pull/4/commits/71d6125880708bb1f0d957f160b912ab5669219a)

Comment (ctyano) [reply .github/workflows/docker-build-publish.yaml:27]: @WindzCUHK Do you mean that all the steps using `${{ github.event_name != 'pull_request' }}` are not final to diverge with a condition if it's a PR or not? And do we actually have any plans to change them in future?
I was wondering the benefit of switching every part as a single variable.

Comment (WindzCUHK) [reply .github/workflows/docker-build-publish.yaml:27]: @ctyano 
> Do you mean that all the steps using ${{ github.event_name != 'pull_request' }} are not final to diverge with a condition if it's a PR or not?

Yes, those steps just depends on whether the container image is pushed or not.
However, there are multiple factors to determine whether the pipeline should/can push or not.

| trigger event | PR src => dest   | push dest     | push |
|---------------|------------------|---------------|------|
| tag           |                  | docker/athenz | ✅    |
| pull_request  | athenz => athenz | docker/athenz | ❌    |
| pull_request  | fork => athenz   | docker/athenz | ❌    |
| pull_request  | fork => athenz   | ghcr/fork     | ✅    |
| pull_request  | fork => fork     | ghcr/fork     | ✅    |

> And do we actually have any plans to change them in future?

Currently, no but I think we should reconsider what should we support in the fork use cases, or just to ignore fork use cases from the pipeline.

Comment (WindzCUHK) [reply .github/workflows/docker-build-publish.yaml:27]: All PRs should not push to the registry. (for development use case, please reverse the related changes before making a PR.)

**Comment by WindzCUHK** [comment]
merged: https://github.com/AthenZ/k8s-athenz-sia/pull/16

**Pull request closed without merging**